packages/web-app-external/src/App.vue

```diff
@@ -1,160 +1,150 @@
+  <iframe
+    v-if="appUrl && method === 'GET'"
+    :src="appUrl"
+    class="oc-width-1-1 oc-height-1-1"
+    :title="iFrameTitle"
+    allowfullscreen
+  />
+  <div v-if="appUrl && method === 'POST' && formParameters" class="oc-height-1-1">
+    <form :action="appUrl" target="app-iframe" method="post">
+      <input ref="subm" type="submit" :value="formParameters" class="oc-hidden" />
+      <div v-for="(item, key, index) in formParameters" :key="index">
+        <input :name="key" :value="item" type="hidden" />
+      </div>
+    </form>
     <iframe
-      v-if="appUrl && method === 'GET'"
-      :src="appUrl"
+      name="app-iframe"
       class="oc-width-1-1 oc-height-1-1"
       :title="iFrameTitle"
       allowfullscreen
     />
-    <div v-if="appUrl && method === 'POST' && formParameters" class="oc-height-1-1">
-      <form :action="appUrl" target="app-iframe" method="post">
-        <input ref="subm" type="submit" :value="formParameters" class="oc-hidden" />
-        <div v-for="(item, key, index) in formParameters" :key="index">
-          <input :name="key" :value="item" type="hidden" />
-        </div>
-      </form>
-      <iframe
-        name="app-iframe"
-        class="oc-width-1-1 oc-height-1-1"
-        :title="iFrameTitle"
-        allowfullscreen
-      />
-    </div>
-  </main>
+  </div>
 </template>
 
 <script lang="ts">
 import { stringify } from 'qs'
-import { mapGetters } from 'vuex'
-import { computed, defineComponent, unref } from 'vue'
+import { PropType, computed, defineComponent, unref, nextTick, ref, watch, VNodeRef } from 'vue'
+import { useTask } from 'vue-concurrency'
+import { useGettext } from 'vue3-gettext'
+
+import { Resource } from 'web-client/src'
 import { urlJoin } from 'web-client/src/utils'
-import AppTopBar from 'web-pkg/src/components/AppTopBar.vue'
-import { queryItemAsString, useAppDefaults, useRouteQuery } from 'web-pkg/src/composables'
+import { queryItemAsString, useRequest, useRouteQuery, useStore } from 'web-pkg/src/composables'
 import { configurationManager } from 'web-pkg/src/configuration'
-import ErrorScreen from './components/ErrorScreen.vue'
-import LoadingScreen from './components/LoadingScreen.vue'
 
 export default defineComponent({
   name: 'ExternalApp',
-  components: {
-    AppTopBar,
-    ErrorScreen,
-    LoadingScreen
+  props: {
+    resource: { type: Object as PropType<Resource>, required: true }
   },
-  setup() {
+  setup(props) {
+    const language = useGettext()
+    const store = useStore()
+
+    const { $gettext, interpolate: $gettextInterpolate } = language
+    const { makeRequest } = useRequest()
+
     const appName = useRouteQuery('app')
+    const appUrl = ref()
+    const formParameters = ref({})
+    const method = ref()
+    const subm: VNodeRef = ref()
+
+    const capabilities = computed(() => store.getters['capabilities'])
+    // FIXME: Make available to appDefaults composable for pageTitle
```

Comment:
FIXME

Reply:
We should instead make sure to go into an error state...

Reply:
Done